src/mds/FSMap.h

```diff
@@ -259,17 +259,21 @@ class FSMap {
       return;
     }
 
-    for (auto &f : filesystems) {
-      std::string_view fs_name = f.second->mds_map.get_fs_name();
+    for(auto it = filesystems.begin(); it != filesystems.end();) {
+      std::string_view fs_name = (*it).second->mds_map.get_fs_name();
       if (std::find(allowed.begin(), allowed.end(), fs_name) == allowed.end()) {
-	filesystems.erase(f.first);
+        it = filesystems.erase(it);
+      } else {
+        ++it;
       }
     }
 
-    for (auto r : mds_roles) {
-      std::string_view fs_name = fs_name_from_gid(r.first);
+    for(auto it = mds_roles.begin(); it != mds_roles.end();) {
+      std::string_view fs_name = fs_name_from_gid((*it).first);
```

Comment:
```suggestion
      std::string_view fs_name = fs_name_from_gid(it->first);
```

Reply:
This PR is inter-linked with https://github.com/ibm-s390-cloud/ceph/commit/3ae4069a9f7586db50dc659e98420af36015113d, right? Might be good to refresh this PR with it as @kotreshhr mentioned [above](https://github.com/ceph/ceph/pull/46749/commits/0e941b351e13f7b409303e7b5215d83410e7a3e2#r902640832)?